crates/xlinectl/src/command/lock.rs

```diff
@@ -15,9 +17,79 @@ pub(crate) async fn execute(client: &mut Client, matches: &ArgMatches) -> Result
     let mut xutex = Xutex::new(client.lock_client(), prefix, None, None).await?;
 
     let xutex_guard = xutex.lock_unsafe().await?;
-    println!("{}", xutex_guard.key());
-    signal::ctrl_c().await.expect("failed to listen for event");
+
+    let exec_command = matches.get_many::<String>("exec_command");
+    let mut should_wait_for_ctrl_c = true;
+
+    if let Some(exec_command) = exec_command {
+        let exec_command_vec: Vec<&String> = exec_command.collect();
+        if !exec_command_vec.is_empty() {
+            let output = execute_exec_command(&exec_command_vec);
+            println!("{output}");
+            should_wait_for_ctrl_c = false;
+        }
+    }
+    if should_wait_for_ctrl_c {
+        println!("{}", xutex_guard.key());
+        signal::ctrl_c().await.expect("failed to listen for event");
+        println!("releasing the lock, ");
+    }
     // let res = lock_resp.unlock().await;
-    println!("releasing the lock, ");
     Ok(())
 }
+
+/// Execute an exec command
+fn execute_exec_command(command_and_args: &[&String]) -> String {
+    let (command, args) = command_and_args
+        .split_first()
+        .expect("Expected at least one exec command");
+    let output = StdCommand::new(command)
+        .args(args)
+        .output()
+        .expect("failed to execute command");
+    String::from_utf8_lossy(&output.stdout).trim().to_owned()
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    struct TestCase {
+        exec_command: Vec<&'static str>,
+        expected_output: String,
+    }
+
+    impl TestCase {
+        fn new(exec_command: Vec<&'static str>, expected_output: String) -> TestCase {
+            TestCase {
+                exec_command,
+                expected_output,
+            }
+        }
+
+        fn run_test(&self) {
+            let exec_command_owned: Vec<String> =
+                self.exec_command.iter().map(ToString::to_string).collect();
+            let exec_command_refs: Vec<&String> = exec_command_owned.iter().collect();
+            let output = execute_exec_command(&exec_command_refs);
+
+            assert_eq!(
+                output.trim(),
+                self.expected_output.trim(),
+                "Failed executing {:?}. Expected output: {}, got: {}",
+                self.exec_command,
+                self.expected_output,
+                output
+            );
+        }
+    }
+
+    #[test]
+    fn test_execute_exec_command() {
+        let test_cases = vec![TestCase::new(vec!["echo", "success"], "success".to_owned())];
+
```

Comment:
You can add more test cases here, maybe 2 ~ 3 ?  When you finish, you can pass all the CI tests.

Reply:
@CrystalAnalyst I have made the changes and squashed my commits as well